second attempt

danchikSlaziet opened the pull request:
all changes in one pullRequest
1) роут /signout удаляет куку, не знаю почему у вас не работает
2) для продакшн сборки секретный ключ jwt хранился в .env, для не продакш у меня простой ключ прям в коде
3) файлы логов у меня итак не добавлялись в репозиторий
4) все роуты итак подключены в index.js и оттуда единый роут подключается в app.js

danchikSlaziet committed:
second attempt

all changes in one pullRequest

dizelgerda reviewed (commented):
# Резюме

Для получения зачета необходимо набрать 80 и больше баллов. Прошу вас не закрывать этот pull request до получения зачета. При работе над ошибками, новый pull request создавать ненужно — загружайте исправления в этот.

**⚠️ Не удаляйте это резюме.** Удаление усложнит и замедлит повторные проверки.

## Backend

### Критерии, влияющие на работоспособность

- [ ] <!--3.64--> В репозитории есть все необходимые инфраструктурные файлы:
  - [x] в проекте есть package.json;
  - [x] .editorconfig;
  - [x] .eslintrc, расширяющий конфигурацию airbnb-base, а также необходимые для работы линтера dev-зависимости;
  - [ ] .gitignore.
  - [x] в .eslintrc добавлено исключение для _id. Директивы eslint-disable, eslint-disable-line и eslint-disable-next-line запрещены.
- [x] <!--3.64--> Нет ошибок линтера. При выполнении команды npx eslint . они должны отсутствовать.
- [x] <!--3.64--> В разделе scripts файла package.json есть команды:
  - [x] npm run start, которая запускает сервер на localhost:3000;
  - [x] npm run dev, которая запускает сервер на localhost:3000 с хот релоудом.
- [ ] <!--21.81--> Описанные роуты работают корректно:
  - [x] запрос на `GET /users/me` возвращает информацию о пользователе (email и имя);
  - [x] `PATCH /users/me` — обновляет информацию о пользователе;
  - [ ] `GET /movies` — все сохранённые **пользователем** фильмы;
  - [x] `POST /movies` — создаёт фильм с переданными в теле данными;
  - [x] `DELETE /movies/movieId` — удаляет сохранённый фильмы по `_id`;
  - [x] `POST /signup` — создаёт пользователя с переданными в теле данными;
  - [x] `POST /signin` — возвращает JWT, если в теле запроса переданы правильные почта и пароль.
  - [ ] Если вы сохраняете JWT в куках, роут `/signout` должен удалять JWT из куки.
- [x] <!--3.64--> Пользователь не может удалить сохранённый фильм из профиля другого пользователя.
- [x] <!--3.64--> Все роуты, кроме /signin и /signup, защищены авторизацией.
- [x] <!--3.64--> Роуты пользователей и роуты фильмов описаны в отдельных файлах.
- [x] <!--7.28--> Ошибки API обрабатываются:
  - [x] используются статус-коды ошибок: 400, 401, 403, 404, 409, 500;
  - [x] если в запросе что-то не так, сервер возвращает статус ответа, который соответствует типу ошибки;
  - [x] сообщение об ошибке соответсвует ее типу;
  - [x] асинхронные обработчики завершены блоком catch;
  - [x] API не возвращает стандартных ошибок базы данных или Node.js.
- [x] <!--3.64--> Реализовано бережное хранение пароля:
  - [x] пароль хранится в виде хеша;
  - [x] API не возвращает хеш пароля клиенту.
- [x] <!--7.28--> Валидация реализована верно:
  - [x] Запрос валидируется перед передачей контроллеру. Тело и, где необходимо, заголовки и параметры проверяются по соответствующим схемам. Если запрос не соответствует схеме, обработка не передаётся контроллеру, и клиент получает ошибку валидации. 
  - [x] Данные валидируются перед добавлением в базу. 
- [x] <!--7.28--> В production-режиме адрес базы данных берётся из process.env.
- [x] <!--3.64--> К серверу можно обратиться по https по адресу домена, указанному в README.md. 
- [ ] <!--7.28--> Правильно реализовано хранение секретного ключа для создания JWT:
  - [ ] для продакшн сборки он хранится в .env файле — этот файл не добавляется в git;
  - [x] в режиме разработки (при process.env.NODE_ENV !== 'production') код запускается и работает и без .env файла.

### Хорошие практики

- [ ] <!--2.14--> Все роуты подключены в файле index.js, который находится в папке routes. Оттуда единый роут подключается в файле app.js.
- [x] <!--2.14--> Асинхронные операции реализованы промисами или async/await.
- [x] <!--2.14--> Валидация описана в отдельном модуле.
- [ ] <!--2.14--> Настроено логгирование:
  - [x] запросы и ответы записываются в файл request.log;
  - [x] ошибки записываются в файл error.log;
  - [ ] файлы логов не добавляются в репозиторий.
- [x] <!--2.14--> Ошибки обрабатываются централизованным обработчиком.
- [x] <!--2.14--> Централизованный обработчик ошибок описан в отдельном модуле.
- [x] <!--2.85--> API размещено на отдельном поддомене, например api.movies-explorer.nomoreparties.co или API приложения располагается на домене вида: name.zone/api, а не просто name.zone

### Рекомендации

- [x] <!--0.71--> Для ошибок API созданы классы, расширяющие конструктор Error.
- [ ] <!--0.71--> Используется модуль Helmet для установки заголовков, связанных с безопасностью.
- [ ] <!--0.71--> Конфигурация и константы в отдельных файлах:
  - [ ] адрес Mongo-сервера в режиме разработки вынесен в отдельный конфигурационный файл;
  - [ ] сообщения ответов и ошибок (константы приложения) вынесены в отдельный файл с константами.
- [ ] <!--0.71--> Настроен rate limiter: число запросов с одного IP в единицу времени ограничено.
- [ ] <!--0.71--> Rate limiter сконфигурирован в отдельном файле и импортируется в app.js.
- [x] <!--0.71--> Для имени базы данных используется `bitfilmsdb`.

## Количество баллов: 60

danchikSlaziet committed:
added gitignore

danchikSlaziet committed:
исправления

1) роут /signout удаляет куку, не знаю почему у вас не работает
2) для продакшн сборки секретный ключ jwt хранился в .env, для не продакш у меня простой ключ прям в коде
3) файлы логов у меня итак не добавлялись в репозиторий
4) все роуты итак подключены в index.js и оттуда единый роут подключается в app.js

dizelgerda reviewed (approved):
# Резюме

Для получения зачета необходимо набрать 80 и больше баллов. Прошу вас не закрывать этот pull request до получения зачета. При работе над ошибками, новый pull request создавать ненужно — загружайте исправления в этот.

**⚠️ Не удаляйте это резюме.** Удаление усложнит и замедлит повторные проверки.

## Backend

### Критерии, влияющие на работоспособность

- [x] <!--3.64--> В репозитории есть все необходимые инфраструктурные файлы:
  - [x] в проекте есть package.json;
  - [x] .editorconfig;
  - [x] .eslintrc, расширяющий конфигурацию airbnb-base, а также необходимые для работы линтера dev-зависимости;
  - [x] .gitignore.
  - [x] в .eslintrc добавлено исключение для _id. Директивы eslint-disable, eslint-disable-line и eslint-disable-next-line запрещены.
- [x] <!--3.64--> Нет ошибок линтера. При выполнении команды npx eslint . они должны отсутствовать.
- [x] <!--3.64--> В разделе scripts файла package.json есть команды:
  - [x] npm run start, которая запускает сервер на localhost:3000;
  - [x] npm run dev, которая запускает сервер на localhost:3000 с хот релоудом.
- [x] <!--21.81--> Описанные роуты работают корректно:
  - [x] запрос на `GET /users/me` возвращает информацию о пользователе (email и имя);
  - [x] `PATCH /users/me` — обновляет информацию о пользователе;
  - [x] `GET /movies` — все сохранённые **пользователем** фильмы;
  - [x] `POST /movies` — создаёт фильм с переданными в теле данными;
  - [x] `DELETE /movies/movieId` — удаляет сохранённый фильмы по `_id`;
  - [x] `POST /signup` — создаёт пользователя с переданными в теле данными;
  - [x] `POST /signin` — возвращает JWT, если в теле запроса переданы правильные почта и пароль.
  - [x] Если вы сохраняете JWT в куках, роут `/signout` должен удалять JWT из куки.
- [x] <!--3.64--> Пользователь не может удалить сохранённый фильм из профиля другого пользователя.
- [x] <!--3.64--> Все роуты, кроме /signin и /signup, защищены авторизацией.
- [x] <!--3.64--> Роуты пользователей и роуты фильмов описаны в отдельных файлах.
- [x] <!--7.28--> Ошибки API обрабатываются:
  - [x] используются статус-коды ошибок: 400, 401, 403, 404, 409, 500;
  - [x] если в запросе что-то не так, сервер возвращает статус ответа, который соответствует типу ошибки;
  - [x] сообщение об ошибке соответсвует ее типу;
  - [x] асинхронные обработчики завершены блоком catch;
  - [x] API не возвращает стандартных ошибок базы данных или Node.js.
- [x] <!--3.64--> Реализовано бережное хранение пароля:
  - [x] пароль хранится в виде хеша;
  - [x] API не возвращает хеш пароля клиенту.
- [x] <!--7.28--> Валидация реализована верно:
  - [x] Запрос валидируется перед передачей контроллеру. Тело и, где необходимо, заголовки и параметры проверяются по соответствующим схемам. Если запрос не соответствует схеме, обработка не передаётся контроллеру, и клиент получает ошибку валидации. 
  - [x] Данные валидируются перед добавлением в базу. 
- [x] <!--7.28--> В production-режиме адрес базы данных берётся из process.env.
- [x] <!--3.64--> К серверу можно обратиться по https по адресу домена, указанному в README.md. 
- [x] <!--7.28--> Правильно реализовано хранение секретного ключа для создания JWT:
  - [x] для продакшн сборки он хранится в .env файле — этот файл не добавляется в git;
  - [x] в режиме разработки (при process.env.NODE_ENV !== 'production') код запускается и работает и без .env файла.

### Хорошие практики

- [ ] <!--2.14--> Все роуты подключены в файле index.js, который находится в папке routes. Оттуда единый роут подключается в файле app.js.
- [x] <!--2.14--> Асинхронные операции реализованы промисами или async/await.
- [x] <!--2.14--> Валидация описана в отдельном модуле.
- [x] <!--2.14--> Настроено логгирование:
  - [x] запросы и ответы записываются в файл request.log;
  - [x] ошибки записываются в файл error.log;
  - [x] файлы логов не добавляются в репозиторий.
- [x] <!--2.14--> Ошибки обрабатываются централизованным обработчиком.
- [x] <!--2.14--> Централизованный обработчик ошибок описан в отдельном модуле.
- [x] <!--2.85--> API размещено на отдельном поддомене, например api.movies-explorer.nomoreparties.co или API приложения располагается на домене вида: name.zone/api, а не просто name.zone

### Рекомендации

- [x] <!--0.71--> Для ошибок API созданы классы, расширяющие конструктор Error.
- [ ] <!--0.71--> Используется модуль Helmet для установки заголовков, связанных с безопасностью.
- [ ] <!--0.71--> Конфигурация и константы в отдельных файлах:
  - [ ] адрес Mongo-сервера в режиме разработки вынесен в отдельный конфигурационный файл;
  - [ ] сообщения ответов и ошибок (константы приложения) вынесены в отдельный файл с константами.
- [ ] <!--0.71--> Настроен rate limiter: число запросов с одного IP в единицу времени ограничено.
- [ ] <!--0.71--> Rate limiter сконфигурирован в отдельном файле и импортируется в app.js.
- [x] <!--0.71--> Для имени базы данных используется `bitfilmsdb`.

## Количество баллов: 95
